### PR TITLE
Task-E 

### PR DESCRIPTION
**Pull request by syedomarazam**
Added new functionality as part of Task-E, its an enhancement to the simulator wherein we performed potential code optimization and removed redundant code from all four components. We created a new header file called code_optimize.hpp which contains commonly used code by all four components. We now use only two variables to specify input and output file paths. set_file_name() function takes cares of this initialization based on the called component. There are several other changes to satisfy good coding practices. Un-used code is commented out and kept for future use

### CODE REVIEW

File: include/transform_output/code_optimize.hpp [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    static std::ostream& sink() {          
+    	return out_data;
+    }
+};
+
+/**
+ * This following section of code is redundant in all four
+ * components and un-used. So we commented out this section.
+ * Please do not remove it as it may serve as a reference in futire.
+ * Its a good coding practice to keep the old code commented as
```
Comment (cruizm) [inline include/transform_output/code_optimize.hpp]: It is not a good practice to keep old code commented. Deprecated code should be removed.
That is why you use github and proper name for the commits. If you need old code, you go back to the commit where it was removed and you take it.

File: test/src/main_receiver.cpp [hunk 2]
```diff
@@ -9,7 +9,8 @@
  *
  */
 
-
+//commented out code is redundant and is moved to code_otpimized.hpp file
```
Comment (cruizm) [inline test/src/main_receiver.cpp]: You do not leave old code commented. You delete it.

**Comment by cruizm** [comment]
You want to check your git ignore. There are .o and .exe files that shouldn't be there

**Comment by cruizm** [comment]
@mohamedali5 I cannot see here the improvement you mentioned. I can only see the "code cleaning"

**Comment by shubhamagrawal6629** [comment]
What was the improvement they proposed?

**Comment by shubhamagrawal6629** [comment]
Your code is not generating the output files it is just using the older files that have been pushed so if you delete a output file and run the program it does generate any output

**Comment by shubhamagrawal6629** [comment]
does not generate any output file

File: include/transform_output/code_optimize.hpp [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+void set_file_name(std::string module_name){
+
+    if(module_name == "subnet"){
+       
+        //#define input_file_name "test/data/subnet_test_output.txt"
+        //#define output_file_name "test/data/subnet_test_output_transform.txt"
+        input_file_name = "test/data/subnet_test_output.txt";
+        output_file_name = "test/data/subnet_test_output_transform.txt";
+        printf("subnet files assigned %s \t %s\n",input_file_name,output_file_name);
```
Comment (p-sky) [inline include/transform_output/code_optimize.hpp]: This line exceeds 80 characters. There are other lines that exceed the limit. Please check all.

File: include/transform_output/code_optimize.hpp [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+struct oss_sink_provider{
+    static std::ostream& sink() {          
+    	return out_data;
+    }
+};
+
+/**
+ * This following section of code is redundant in all four
+ * components and un-used. So we commented out this section.
+ * Please do not remove it as it may serve as a reference in futire.
```
Comment (p-sky) [inline include/transform_output/code_optimize.hpp]: The word future is mispelled as futire. Please correct.

File: include/transform_output/code_optimize.hpp [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+                                                 debug, 
+                                                 state, 
+                                                 log_messages, 
+                                                 routing, 
+                                                 global_time, 
+                                                 local_time>;
+ */
+
+/**
+ * Loggers definition of cadmium to call source loggers strcture
```
Comment (p-sky) [inline include/transform_output/code_optimize.hpp]: The word strcture is missing 'u' letter

**Comment by shubhamagrawal6629** [comment]
> does not generate any output file

Sorry my bad there is no such problem

**Review by p-sky — APPROVED** [review]

**Comment by syedomarazam** [comment]
This pull request is not needed anymore. We merged Task#E changes to master through another pull request. See below https://github.com/mohamedali5/AlternateBitProtocolSimulator/pull/8

**Comment by syedomarazam** [comment]
This pull request is not needed anymore. We merged Task#E changes to master through another pull request. See below #8

**Pull request closed without merging**